RetroBar/Utilities/HotkeyManager.cs

```diff
@@ -0,0 +1,198 @@
+﻿using ManagedShell.Common.Helpers;
+using ManagedShell.Common.Logging;
+using ManagedShell;
+using ManagedShell.Common.SupportingClasses;
+using ManagedShell.Interop;
+using ManagedShell.ShellFolders;
+using ManagedShell.WindowsTasks;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Windows.Data;
+using System.Windows.Forms;
+using System.Windows.Threading;
+
+namespace RetroBar.Utilities
+{
+    public class HotkeyManager : IDisposable
+    {
+        public HotkeyListenerWindow listenerWindow;
+
+        public HotkeyManager()
+        {
+            listenerWindow = new HotkeyListenerWindow();
+            
+            if (Settings.Instance.OverrideHotkeys) listenerWindow.RegisterHotkeys();
+            
+            Settings.Instance.PropertyChanged += Settings_PropertyChanged;
+        }
+
+        public void Dispose()
+        {
+            if (Settings.Instance.OverrideHotkeys) listenerWindow.UnregisterHotkeys();
+
+            listenerWindow?.Dispose();
+        }
+
+
+        #region Events
+        public class TaskbarHotkeyEventArgs : EventArgs
+        {
+            public int index;
+        }
+
+        public static event EventHandler<TaskbarHotkeyEventArgs> TaskbarHotkeyPressed;
+
+        private void Settings_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == nameof(Settings.OverrideHotkeys))
+            {
+                if (Settings.Instance.OverrideHotkeys) listenerWindow.RegisterHotkeys();
+                else listenerWindow.UnregisterHotkeys();
+            }
+        }
+        #endregion
+
+
+        #region Explorer hotkey table search stuff
+
+        static readonly byte[] explorerHotkeySearchPattern = { // 4 entries should be enough to uniquely identify the hotkey table
+                0x31,0x00,0x00,0x00, 0x08,0x00,0x00,0x00,
+                0x32,0x00,0x00,0x00, 0x08,0x00,0x00,0x00,
+                0x33,0x00,0x00,0x00, 0x08,0x00,0x00,0x00,
+                0x34,0x00,0x00,0x00, 0x08,0x00,0x00,0x00,
+            };
+
+        const int explorerHotkeyIndexDefault = 17; // Current index as of 24H2, used as a backup in case the index can't be determined
+
+        // Finds the index of the first taskbar hotkey in explorer's hotkey table
+        // See https://github.com/dremin/RetroBar/pull/1128#issuecomment-2802222222 for more info
+        private static int FindExplorerTaskbarHotkeyIndex()
+        {
```

Comment:
These members should not be static. Instead, to access `TaskbarHotkeyPressed` from `TaskList` and `Toolbar`, the `HotkeyManager` should be plumbed through `WindowManager` and `Taskbar`, then the controls can access it from their `Host` property.